### PR TITLE
DOC: Improve docs for `jax.numpy`: `isfinite`, `isinf`, `isposinf`, `isneginf` and `isnan`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

**Review by jakevdp — CHANGES_REQUESTED** [review]
Thanks!

File: jax/_src/numpy/ufuncs.py [hunk 1]
```diff
@@ -2924,19 +2978,102 @@ def _isposneginf(infinity: float, x: ArrayLike, out) -> Array:
...
+  """Returns a boolean array indicating whether each element of input is ``NaN``.
+
+  JAX implementation of :obj:`numpy.isnan`.
+
+  Args:
+    x: input array or scalar.
+
+  Returns:
+    An boolean array of same shape as ``x``. `True`, if the element of ``x`` is
+    not a number i.e., ``NaN`` and, ``False``, otherwise.
```
[the remark this answers is not in the document]
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Done

File: jax/_src/numpy/ufuncs.py [hunk 2]
```diff
@@ -2924,19 +2978,102 @@ def _isposneginf(infinity: float, x: ArrayLike, out) -> Array:
...
+  Return boolean array indicating whether each element of input is negative infinite.
+
+  JAX implementation of :obj:`numpy.isneginf`.
+
+  Args:
+    x: input array or scalar. ``complex`` dtype are not supported.
+
+  Returns:
+    An boolean array of same shape as ``x``. `True`, if the element of ``x`` is
+    ``-inf`` and, ``False``, otherwise.
```
[the remark this answers is not in the document]
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Done

File: jax/_src/numpy/ufuncs.py [hunk 3]
```diff
@@ -2896,9 +2923,36 @@ def isfinite(x: ArrayLike, /) -> Array:
...
+  """Return a boolean array indicating whether each element of input is infinite.
+
+  JAX implementation of :obj:`numpy.isinf`.
+
+  Args:
+    x: input array or scalar.
+
+  Returns:
+    An boolean array of same shape as ``x``. `True`, if the element of ``x`` is
+    ``inf``, ``-inf`` and, ``False``, otherwise.
```
[the remark this answers is not in the document]
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Done

File: jax/_src/numpy/ufuncs.py [hunk 4]
```diff
@@ -2924,19 +2978,102 @@ def _isposneginf(infinity: float, x: ArrayLike, out) -> Array:
...
+  Return boolean array indicating whether each element of input is positive infinite.
+
+  JAX implementation of :obj:`numpy.isposinf`.
+
+  Args:
+    x: input array or scalar. ``complex`` dtype are not supported.
+
+  Returns:
+    An boolean array of same shape as ``x``. `True`, if the element of ``x`` is
+    ``inf`` and, ``False``, otherwise.
```
[the remark this answers is not in the document]
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Done

File: jax/_src/numpy/ufuncs.py [hunk 5]
```diff
@@ -2883,9 +2883,36 @@ def modf(x: ArrayLike, /, out=None) -> tuple[Array, Array]:
...
+  """Return a boolean array indicating whether each element of input is finite.
+
+  JAX implementation of :obj:`numpy.isfinite`.
+
+  Args:
+    x: input array or scalar.
+
+  Returns:
+    An boolean array of same shape as ``x``. `True`, if the element of ``x`` is
+    not ``inf``, ``-inf`` or ``NaN`` and, ``False``, otherwise.
```
[the remark this answers is not in the document]
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Thanks! Done.

**Review by jakevdp — APPROVED** [review]
Looks great, thanks!

**Pull request merged**